operator/controllers/credential_controller.go

```diff
@@ -0,0 +1,572 @@
+	return &CredentialReconciler{
+		helm:          helm,
+		resourceCache: make(map[kindWithGroup]metav1.APIResource),
+	}
+}
+
+// +kubebuilder:rbac:groups=servicebroker.metabroker.suse.com,resources=credentials,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=servicebroker.metabroker.suse.com,resources=credentials/status,verbs=get;update;patch
+
+const credentialReconcileTimeout = time.Second * 10
+
+// Reconcile reconciles a Credential resource.
+func (r *CredentialReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), credentialReconcileTimeout)
+	defer cancel()
+
+	_ = r.log.WithValues("credential", req.NamespacedName)
+
+	credential := &servicebrokerv1alpha1.Credential{}
+	if err := r.Get(ctx, req.NamespacedName, credential); err != nil {
+		if errors.IsNotFound(err) {
+			// TODO: The credential no longer exists; run any unbinding steps necessary.
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	instance := &servicebrokerv1alpha1.Instance{}
+	if err := r.Get(
+		ctx,
+		types.NamespacedName{
+			Name:      credential.InstanceRef.Name,
+			Namespace: credential.InstanceRef.Namespace,
+		},
+		instance,
+	); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if len(credential.OwnerReferences) == 0 {
+		if err := ctrl.SetControllerReference(instance, credential, r.scheme); err != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.Update(ctx, credential); err != nil {
+			return ctrl.Result{}, err
+		}
+		// TODO: make this a constant.
+		return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
+	}
+
+	secret, err := r.getOrCreateSecret(ctx, &req, credential)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if secret == nil {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	plan := &servicebrokerv1alpha1.Plan{}
+	if err := r.Get(
+		ctx,
+		types.NamespacedName{
+			Name:      instance.Spec.Plan,
+			Namespace: req.Namespace,
+		},
+		plan,
+	); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return r.runBindingPod(ctx, &req, credential, instance, plan, secret.Name)
+}
+
+// getOrCreateSecret returns or creates the existing secret referenced by the credential. When the
+// secret and the error returning values are nil, it means the secret was created. A requeue should
+// be issued for the next reconcile loop to get the current secret from the API server.
+func (r *CredentialReconciler) getOrCreateSecret(
+	ctx context.Context,
+	req *ctrl.Request,
+	credential *servicebrokerv1alpha1.Credential,
+) (*corev1.Secret, error) {
+	desired := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      credential.SecretRef.Name,
+			Namespace: req.Namespace,
+			// TODO: add proper labels.
+		},
+		Data: map[string][]byte{},
+	}
+	if err := ctrl.SetControllerReference(credential, desired, r.scheme); err != nil {
+		return nil, err
+	}
+	current := &corev1.Secret{}
+	if err := r.Get(
```

Comment:
Stylistically, this would probably be easier to read as
```golang
err := r.Get(…)
if err != nil {
…
```
Multi-line conditionals tend to be confusing.  (Of course, in this case we could `if err == nil { return }`, but whether that is easier to read is very subjective.)

Reply:
I think the readability here is a matter of getting used to, honestly.